.cursor/rules/dotcms-unit-test-rules.mdc

```diff
@@ -0,0 +1,274 @@
+---
+description: Unit Test Rules
+globs: core-web/{apps,libs}/**/*.{spec.ts,html,ts,mock.ts}
+---
+# DotCMS Angular Unit Testing Rules
+
+## Tech Stack for Testing
+
+- **Testing Framework**: Jest
+- **Testing Library**: Spectator + @testing-library/angular
+- **Coverage Tool**: Jest Coverage
+- **Mocking**: Jest + ts-mockito
+- **E2E**: Playwrite (when needed)
+
+## File Structure and Naming
+
+- Test files must be named with `.spec.ts` suffix
+- Place test files adjacent to the file being tested
+- Follow pattern: `[name].spec.ts` for all testable files
+- Group related test files in a `__tests__` directory for complex components
+- Create mock files with `.mock.ts` suffix in a `__mocks__` directory
+
+## Element Selection Best Practices
+
+### Using data-testid
+
+1. **Always use data-testid for element selection**
+```html
+<!-- ❌ Don't use class selectors -->
+<button class="submit-button">Submit</button>
+
+<!-- ✅ Use data-testid -->
+<button data-testid="submit-button">Submit</button>
+```
+
+2. **Naming Convention for data-testid**
+```html
+<!-- Components -->
+<div data-testid="user-profile">
+  <!-- Form Controls -->
+  <input data-testid="user-profile-name-input" />
+  <button data-testid="user-profile-save-button">Save</button>
+  
+  <!-- Lists -->
+  <ul data-testid="user-profile-items-list">
+    <li data-testid="user-profile-item-1">Item 1</li>
+  </ul>
+</div>
+```
+
+3. **Finding Elements**
+```typescript
+// ❌ Don't use class selectors
+const button = fixture.debugElement.query(By.css('.submit-button'));
+
+// ✅ Use data-testid
+const button = screen.getByTestId('submit-button');
```

Comment:
@oidacra Actually, in unit tests, we use `spectator.query(byTestId('id'))` or just `byTestId('id')`, but from spectator

Reply:
ohh good catch. Let me add that.

Reply:
Improved